### PR TITLE
[Base] Exception and FileInfo: remove unused includes

### PR DESCRIPTION
**Pull request by donovaly**

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6434">pipeline status</a> for feature branch [PR_6434](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6434). Pipeline [477358543](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/477358543) was triggered at [033dc96](https://github.com/FreeCAD/FreeCAD/pull/6434/commits/033dc962e18a75b344aa6417ad29cbfcc14b5e09). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6434">pipeline status</a> for feature branch [PR_6434](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6434). Pipeline [477358932](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/477358932) was triggered at [3a73608](https://github.com/FreeCAD/FreeCAD/pull/6434/commits/3a73608381657a9e2f50cf2fb898a7a427bde426). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Base/FileTemplate.cpp [hunk 1]
```diff
@@ -23,10 +23,6 @@
 
 #include "PreCompiled.h"
 
-#ifndef _PreComp_
```
Comment (wwmayer) [inline src/Base/FileTemplate.cpp:26]: FileTemplate.cpp serves as template for newly created files to see what's needed. Thus the PreComp stuff shouldn't be removed.

File: src/Base/FileInfo.cpp [hunk 2]
```diff
@@ -27,31 +27,20 @@
 #ifndef _PreComp_
 # include <algorithm>
 # include <cassert>
-# include <cstdio>
-# include <cstdlib>
-# include <fstream>
-# include <climits>
-# include <cstring>
```
Comment (wwmayer) [inline src/Base/FileInfo.cpp:34]: FileInfo.cpp uses strcasecmp that is provided by cstring

File: src/Base/Exception.cpp [hunk 3]
```diff
@@ -24,12 +24,9 @@
 
 #include "PreCompiled.h"
 
-#include <typeinfo>
-
 #include "Exception.h"
 #include "Console.h"
 #include "PyObjectBase.h"
-#include <CXX/Objects.hxx>
```
Comment (wwmayer) [inline src/Base/Exception.cpp:32]: PyCXX classes are used in Exception.cpp

File: src/Base/Exception.cpp [hunk 4]
```diff
@@ -24,12 +24,9 @@
 
 #include "PreCompiled.h"
 
-#include <typeinfo>
```
Comment (wwmayer) [inline src/Base/Exception.cpp:27]: typeid() is used in Exception.cpp

**Comment by wwmayer** [comment]
89b37a5110fbc

File: src/Base/Exception.cpp (hunk 4 above)
Comment (donovaly) [reply src/Base/Exception.cpp:27]: OK, but now you committed it with the deletion: https://github.com/FreeCAD/FreeCAD/commit/89b37a5110fbcc54248e7b9061713b685051442c

File: src/Base/Exception.cpp (hunk 3 above)
Comment (donovaly) [reply src/Base/Exception.cpp:32]: OK, but now you committed it with the deletion: https://github.com/FreeCAD/FreeCAD/commit/89b37a5110fbcc54248e7b9061713b685051442c

Comment (wwmayer) [reply src/Base/Exception.cpp:32]: It's included by PyObjectBase.h

File: src/Base/Exception.cpp (hunk 4 above)
Comment (wwmayer) [reply src/Base/Exception.cpp:27]: It's included by PyObjectBase.h

File: src/Base/Exception.cpp (hunk 3 above)
Comment (donovaly) [reply src/Base/Exception.cpp:32]: OK, therefore I did not get a compilation error with my PR.

**Pull request closed without merging**